Hookup SubjectDescriptor in CommandHandler

mlepage-google opened the pull request:
#### Problem
SubjectDescriptor needs to be hooked up in CommandHandler
so AccessControl check can be performed before executing command.

#### Change overview
Get the SubjectDescriptor from the ExchangeContext/SessionHandle.

#### Testing
Testing is currently an issue, they seem to run CommandHandler
without having an ExchangeContext/SessionHandle.

mlepage-google commented on src/app/CommandHandler.cpp:
```diff
@@ -254,7 +254,12 @@ CHIP_ERROR CommandHandler::ProcessCommandDataIB(CommandDataIB::Parser & aCommand
 
     {
-        Access::SubjectDescriptor subjectDescriptor; // TODO: get actual subject descriptor
+        if (mpExchangeCtx == nullptr || mpExchangeCtx->HasSessionHandle() == false)
```
done

msandstedt commented:
Is there an override to bypass the permissions check?  We can't enable actual ACL checks until our basic flows can actually distribute permissions in a fabric.

I would think at a minimum this means we need the commissioner to allow writing an arbitrary AdminSubject and we need CATs supported as subjects.

mlepage-google commented:
@msandstedt don't worry, ACL won't be turned on until it's ready to use.

mlepage-google commented:
This PR is just to get the actual subject descriptor from the underlying exchange context, which must always be present.

However, we have tests which violate that, by not providing a proper exchange context or mock. This is why CI tests fail on this PR.

Now that I got a few other things done, I'll start to look at getting the tests running with this PR.

mlepage-google commented:
OK, I spent the entire day spelunking through TestCommandInteraction.cpp and I think I've managed to add an exchange context for the test cases that did not have it.

I removed the special check for an exchange context in test cases in CommandHandler::ProcessInvokeRequest and replaced it with a VerifyOrReturnError.

@bzbarsky-apple @yunhanw-google @erjiaqing 
PLEASE CHECK THE DIFF ON MY COMMIT JUST NOW AND TELL ME IF I DIDN'T DO THIS CORRECTLY.

Tests do seem to pass locally (will see how CI goes) and I was still able to read/write attributes using REPL with all-clusters-app on Linux.

bzbarsky-apple commented:
>  PLEASE CHECK THE DIFF ON MY COMMIT JUST NOW AND TELL ME IF I DIDN'T DO THIS CORRECTLY.

As discussed on Slack, seems fine for now.

mlepage-google commented:
Rebase to pull in some fixes for Darwin CI tests.
